File: src/main/java/io/stouder/adonis/edge/parsing/EdgeTokenTypes.java
package io.stouder.adonis.edge.parsing;

import com.intellij.psi.tree.IElementType;
import com.intellij.psi.tree.TokenSet;

public final class EdgeTokenTypes {

    public static final IElementType STATEMENTS = new EdgeCompositeElementType("STATEMENTS");

    public static final IElementType TAG = new EdgeCompositeElementType("TAG");

    public static final IElementType MUSTACHE = new EdgeCompositeElementType("MUSTACHE");
    public static final IElementType COMMENT_MUSTACHE = new EdgeCompositeElementType("COMMENT_MUSTACHE");

    public static final IElementType MUSTACHE_CONTENT = new EdgeElementType("MUSTACHE_CONTENT");
    public static final IElementType COMMENT_MUSTACHE_CONTENT = new EdgeElementType("COMMENT_MUSTACHE_CONTENT");

    public static final IElementType MUSTACHE_OPEN = new EdgeElementType("MUSTACHE_OPEN");
    public static final IElementType MUSTACHE_CLOSE = new EdgeElementType("MUSTACHE_CLOSE");

    public static final IElementType SAFE_MUSTACHE_OPEN = new EdgeElementType("SAFE_MUSTACHE_OPEN");
    public static final IElementType SAFE_MUSTACHE_CLOSE = new EdgeElementType("SAFE_MUSTACHE_CLOSE");

    public static final IElementType ESCAPED_MUSTACHE_OPEN = new EdgeElementType("ESCAPED_MUSTACHE_OPEN");
    public static final IElementType ESCAPED_MUSTACHE_CLOSE = new EdgeElementType("ESCAPED_MUSTACHE_CLOSE");

    public static final IElementType ESCAPED_SAFE_MUSTACHE_OPEN = new EdgeElementType("ESCAPED_SAFE_MUSTACHE_OPEN");
    public static final IElementType ESCAPED_SAFE_MUSTACHE_CLOSE = new EdgeElementType("ESCAPED_SAFE_MUSTACHE_CLOSE");

    public static final IElementType COMMENT_MUSTACHE_OPEN = new EdgeElementType("COMMENT_MUSTACHE_OPEN");
    public static final IElementType COMMENT_MUSTACHE_CLOSE = new EdgeElementType("COMMENT_MUSTACHE_CLOSE");

    public static final IElementType TAG_NAME = new EdgeElementType("TAG_NAME");
    public static final IElementType TAG_CONTENT_OPEN = new EdgeElementType("TAG_CONTENT_OPEN");
    public static final IElementType TAG_CONTENT_CLOSE = new EdgeElementType("TAG_CONTENT_CLOSE");

    public static final IElementType TAG_CONTENT = new EdgeElementType("TAG_CONTENT");
    public static final IElementType NEWLINE = new EdgeElementType("NEWLINE");
    public static final IElementType WHITE_SPACE = new EdgeElementType("WHITE_SPACE");
    public static final IElementType CONTENT = new EdgeElementType("CONTENT");
    public static final IElementType OUTER_ELEMENT_TYPE = new EdgeElementType("OUTER_ELEMENT_TYPE");

    public static final IElementType INVALID = new EdgeElementType("INVALID");

    public static final TokenSet WHITESPACES = TokenSet.create(WHITE_SPACE);
    public static final TokenSet COMMENTS = TokenSet.create();
}

File: src/main/java/io/stouder/adonis/edge/parsing/EdgeTokenTypes.kt
package io.stouder.adonis.edge.parsing

import com.intellij.psi.tree.IElementType
import com.intellij.psi.tree.TokenSet

object EdgeTokenTypes {
    val STATEMENTS: IElementType = EdgeCompositeElementType("STATEMENTS")
    val TAG: IElementType = EdgeCompositeElementType("TAG")
    val MUSTACHE: IElementType = EdgeCompositeElementType("MUSTACHE")
    val COMMENT_MUSTACHE: IElementType = EdgeCompositeElementType("COMMENT_MUSTACHE")

    val MUSTACHE_CONTENT: IElementType = EdgeElementType("MUSTACHE_CONTENT")
    val COMMENT_MUSTACHE_CONTENT: IElementType = EdgeElementType("COMMENT_MUSTACHE_CONTENT")

    val MUSTACHE_OPEN: IElementType = EdgeElementType("MUSTACHE_OPEN")
    val MUSTACHE_CLOSE: IElementType = EdgeElementType("MUSTACHE_CLOSE")

    val SAFE_MUSTACHE_OPEN: IElementType = EdgeElementType("SAFE_MUSTACHE_OPEN")
    val SAFE_MUSTACHE_CLOSE: IElementType = EdgeElementType("SAFE_MUSTACHE_CLOSE")

    val ESCAPED_MUSTACHE_OPEN: IElementType = EdgeElementType("ESCAPED_MUSTACHE_OPEN")
    val ESCAPED_MUSTACHE_CLOSE: IElementType = EdgeElementType("ESCAPED_MUSTACHE_CLOSE")

    val ESCAPED_SAFE_MUSTACHE_OPEN: IElementType = EdgeElementType("ESCAPED_SAFE_MUSTACHE_OPEN")
    val ESCAPED_SAFE_MUSTACHE_CLOSE: IElementType = EdgeElementType("ESCAPED_SAFE_MUSTACHE_CLOSE")

    val COMMENT_MUSTACHE_OPEN: IElementType = EdgeElementType("COMMENT_MUSTACHE_OPEN")
    val COMMENT_MUSTACHE_CLOSE: IElementType = EdgeElementType("COMMENT_MUSTACHE_CLOSE")

    val TAG_NAME: IElementType = EdgeElementType("TAG_NAME")
    val TAG_CONTENT_OPEN: IElementType = EdgeElementType("TAG_CONTENT_OPEN")
    val TAG_CONTENT_CLOSE: IElementType = EdgeElementType("TAG_CONTENT_CLOSE")

    val TAG_CONTENT: IElementType = EdgeElementType("TAG_CONTENT")
    val NEWLINE: IElementType = EdgeElementType("NEWLINE")
    val WHITE_SPACE: IElementType = EdgeElementType("WHITE_SPACE")
    val CONTENT: IElementType = EdgeElementType("CONTENT")
    val OUTER_ELEMENT_TYPE: IElementType = EdgeElementType("OUTER_ELEMENT_TYPE")

    val INVALID: IElementType = EdgeElementType("INVALID")

    val WHITESPACES: TokenSet = TokenSet.create(WHITE_SPACE)
    val COMMENTS: TokenSet = TokenSet.create()
}
